config/config.go

```diff
@@ -289,19 +289,13 @@ func (ch *ConfigHandler) setConfigAndNotify(cfg *Config) {
 
 // mergeResp merges the old and new configuration responses. The merged response is returned
 // along with any error that occurred during the merge.
-func mergeResp(oldConfig, newConfig C.ConfigResponse) (C.ConfigResponse, error) {
-	newOpts := newConfig.Options
-	// we set the options to empty since we want to use the new options. we replace them later
-	oldConfig.Options = option.Options{}
-	newConfig.Options = option.Options{}
-
-	// badjson.Merge ensures a deep copy of the old config is made
-	merged, err := badjson.Merge(context.Background(), oldConfig, newConfig, false)
-	if err != nil {
-		return C.ConfigResponse{}, err
+func mergeResp(oldConfig, newConfig *C.ConfigResponse) (*C.ConfigResponse, error) {
+	oldConfigCopy := reprint.This(*oldConfig).(C.ConfigResponse)
```

Comment:
We could just do this to avoid adding reprint. This does a deep copy
```suggestion
	var oldCopy C.ConfigResponse
	buf, _ := json.Marshal(oldConfig)
	_ = json.Unmarshal(buf, &oldCopy)

```

Reply:
yeah I do generally like the json way of doing it, but reprint's way underneath is slicker

Reply:
I do feel we should try to keep dependencies to a minimum, only introducing them when necessary. But if you prefer we use reprint, then alright.

Reply:
Yeah I generally agree, but this deep copy problem is pretty common, the code for doing it is super minimal if you click through, and I think it's pretty clearly the "right" way to do it I think.

Reply:
I actually just emailed the developer because he also wrote his own VPN in Go and I think Flutter

Reply:
> but this deep copy problem is pretty common

I could have sworn go had a built in deep copy, I think `reflect.DeepCopy`. I spent 20 minutes yesterday searching online for where they moved it..you would think they would include that.